src/taskpane/server.js

```diff
@@ -4188,24 +4582,45 @@ function getFromSpiraExcel(model, fieldTypeEnums) {
   return Excel.run(function (context) {
     var fields = model.fields;
     var sheet = context.workbook.worksheets.getActiveWorksheet(),
-      sheetRange = sheet.getRangeByIndexes(1, 0, EXCEL_MAX_ROWS, fields.length);
+      sheetRange = sheet.getRangeByIndexes(1, 0, MAX_ROWS_PER_PAGE, fields.length);
     sheet.load("name");
     sheetRange.load("values");
-    var requiredSheetName = model.currentArtifact.name + ", PR-" + model.currentProject.id;
+    var requiredSheetName;
+
+    if (model.currentOperation) {
+      //administrator mode
+
+      var operation = model.operations.filter(function (operation) {
+        return operation.id == model.currentOperation;
+      })[0];
+
+      if (operation.type == "send-system") {
+        //system wide operations
+        requiredSheetName = model.currentArtifact.name + ", system";
+
+      }
+      else if (operation.type == "send-template" || operation.type == "get-template") {
+        //template-based operations
+        requiredSheetName = model.currentArtifact.name + ", TP-" + model.currentTemplate.id;
+      }
+    }
+    else {
+      //stardard artifact functions
+      requiredSheetName = model.currentArtifact.name + ", PR-" + model.currentProject.id;
+    }
+

```

Comment:
This logic seems repeated 5 times in the code - lines 1172, 1531, 2170, 4400, could this not be genericized for these cases?

Reply:
Indeed, I create the function matchSheetName to handle it. Good catch!